### PR TITLE
[#1872114, #1896944] Implement app service action contributor base on new action framework

### PR DESCRIPTION
**Pull request by Flanker32**
Implement app service action contributor base on new action framework.

[AB#1872114](https://dev.azure.com/mseng/a4d27ce2-a42d-4b71-8eef-78cee9a9728e/_workitems/edit/1872114)

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-appservice/src/main/java/com/microsoft/azure/toolkit/intellij/appservice/AppServiceIntelliJActionsContributor.java [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+    @Override
+    public void registerActions(AzureActionManager am) {
+        final BiConsumer<AppServiceFile, AnActionEvent> openFileHandler = (file, e) -> AzureTaskManager
+                .getInstance().runLater(() -> new AppServiceFileAction().openAppServiceFile(file, e.getProject()));
+        final ActionView.Builder openFileView = new ActionView.Builder("Open File", null)
+                .title(s -> Optional.ofNullable(s)
+                        .map(r -> AzureString.format("appservice|file.download", ((AppServiceFile) r).getName()))
+                        .orElse(null))
+                .enabled(s -> s instanceof AppServiceFile);
+        am.registerAction(AppServiceFileActionsContributor.APP_SERVICE_FILE_VIEW, new Action<>(openFileHandler, openFileView));
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-appservice/src/main/java/com/microsoft/azure/toolkit/intellij/appservice/AppServiceIntelliJActionsContributor.java:62]: keep action id as action + <operation id>

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-appservice/src/main/java/com/microsoft/azure/toolkit/intellij/appservice/AppServiceIntelliJActionsContributor.java [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+        am.registerHandler(FunctionAppActionsContributor.TRIGGER_FUNCTION, triggerPredicate, triggerFunctionHandler);
+    }
+
+    @Override
+    public int getOrder() {
+        return INITIALIZE_ORDER;
+    }
+
+    @RequiredArgsConstructor
+    static class TriggerRequest {
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-appservice/src/main/java/com/microsoft/azure/toolkit/intellij/appservice/AppServiceIntelliJActionsContributor.java:151]: consider move TriggerRequest to triggerFunction

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-appservice/src/main/java/com/microsoft/azure/toolkit/lib/legacy/function/FunctionCoreToolsCombobox.java [hunk 3]
```diff
@@ -94,6 +99,12 @@ public FunctionCoreToolsCombobox(Project project, boolean includeSettings) {
         }
     }
 
+    private void openAzureSettingsPanel() {
+        final Action<Void> openSettingsAction = AzureActionManager.getInstance().getAction(ResourceCommonActionsContributor.OPEN_AZURE_SETTINGS);
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-appservice/src/main/java/com/microsoft/azure/toolkit/lib/legacy/function/FunctionCoreToolsCombobox.java:103]: add todo to await action

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-lib/src/main/java/com/microsoft/azure/toolkit/intellij/common/AzureFileType.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+import com.intellij.openapi.fileTypes.FileType;
+import com.intellij.openapi.vfs.VirtualFile;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import javax.swing.*;
+import java.nio.charset.StandardCharsets;
+
+public class AzureFileType implements FileType {
```
Comment (andxu) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-lib/src/main/java/com/microsoft/azure/toolkit/intellij/common/AzureFileType.java:16]: Check duplicate class

**Review by andxu — APPROVED** [review]
LGTM

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-lib/src/main/java/com/microsoft/azure/toolkit/intellij/common/AzureFileType.java (hunk 4 above)
Comment (Flanker32) [reply PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-lib/src/main/java/com/microsoft/azure/toolkit/intellij/common/AzureFileType.java:16]: Will resolve in later PR

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-appservice/src/main/java/com/microsoft/azure/toolkit/lib/legacy/function/FunctionCoreToolsCombobox.java (hunk 3 above)
Comment (Flanker32) [reply PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-appservice/src/main/java/com/microsoft/azure/toolkit/lib/legacy/function/FunctionCoreToolsCombobox.java:103]: We need to refactor the action framework for execute sync, add work item to track it

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-appservice/src/main/java/com/microsoft/azure/toolkit/intellij/appservice/AppServiceIntelliJActionsContributor.java (hunk 2 above)
Comment (Flanker32) [reply PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-appservice/src/main/java/com/microsoft/azure/toolkit/intellij/appservice/AppServiceIntelliJActionsContributor.java:151]: Will refactor it after refactor for app service library

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-appservice/src/main/java/com/microsoft/azure/toolkit/intellij/appservice/AppServiceIntelliJActionsContributor.java (hunk 1 above)
Comment (Flanker32) [reply PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-appservice/src/main/java/com/microsoft/azure/toolkit/intellij/appservice/AppServiceIntelliJActionsContributor.java:62]: Will resolve in later PR

**Pull request merged**